### PR TITLE
Jon/home icon

### PR DESCRIPTION
**Pull request by jphannifan**
The home button now indicates the home status of the printer. When the printer is not homed, the icon displays with a red tint, and when all axes are homed, the icon displays with a green tint.

When the printer is left idle for long enough, the steppers are disabled and the homing status resets. I have tested this and when the steppers are turned off, the indicator updates as it should.

I've also added the warning dialog that displays when the axis controls are used and the printer is not homed. There is a temporary ignore for this implemented. 

Not homed:
image

Homed:
image

### CODE REVIEW

File: bbl_screen-patch/patches/printerui/qml/printer/AdjustPage.qml.patch [hunk 1]
```diff
@@ -1,32 +1,101 @@
...
-             }
+@@ -873,11 +889,13 @@
+         dialogStack.popupDialog(
+                     "TextConfirm", {
+                         name: "Limit Frame",
+-                        titles: [qsTr("Home"), qsTr("Close")],
+-                        text: qsTr("Before any movement, please home all axes to find the boundary of build volume."),
++                        type: TextConfirm.YES_NO_CANCEL,
++                        titles: [qsTr("Home"),qsTr("Hide"), qsTr("Close")],
++                        text: qsTr("Before any movement, please home all axes to find the boundary of build volume. To disable this message temporarily, press 'Hide'."),
```
[the remark this answers is not in the document]
Comment (jphannifan) [reply bbl_screen-patch/patches/printerui/qml/printer/AdjustPage.qml.patch]: Yea I got a bit lazy here. Thanks for spotting that

**Comment by bani6809** [comment]
indicating status via color changes may be an accessability issue for visually impaired users.

consider a solid filled in home icon vs an outlined home icon as a better indicator of home status.

**Comment by jphannifan** [comment]
> indicating status via color changes may be an accessability issue for visually impaired users.
> 
> consider a solid filled in home icon vs an outlined home icon as a better indicator of home status.

It is a solid fill though. The color of the entire image (the house icon) is changed, not just the outline. We can show a text-based indicator somewhere too, but I don't know where to put that just yet.

**Review by jwise — APPROVED** [review]
I have not tested this but it looks correct, which is the best kind of correct.

Agree that it would be nice to do a little extra a11y stuff on this sooner or later, but I don't think that needs to block this going in.  #174 filed for that.

**Comment by bani6809** [comment]
> > indicating status via color changes may be an accessability issue for visually impaired users.
> > consider a solid filled in home icon vs an outlined home icon as a better indicator of home status.
> 
> It is a solid fill though. The color of the entire image (the house icon) is changed, not just the outline. We can show a text-based indicator somewhere too, but I don't know where to put that just yet.

What I mean is something like this:

red, silhouette = unhomed

image

green, solid = homed

image

**Comment by jphannifan** [comment]
Here's the icon with the opacity increased. I don't think we need text on it either
image
image

**Comment by bani6809** [comment]
yeah that works as long as unhomed is a red silhouette.

image

image

**Pull request merged**